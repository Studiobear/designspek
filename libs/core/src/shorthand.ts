import { cap } from './util'

const m = 'margin'
const p = 'padding'
const t = 'top'
const r = 'right'
const b = 'bottom'
const l = 'left'

export const shortHandAttr = new Map([
  ['m', [m]],
  ['mt', [`${m}${cap(t)}`]],
  ['mr', [`${m}${cap(r)}`]],
  ['mb', [`${m}${cap(b)}`]],
  ['ml', [`${m}${cap(l)}`]],
  ['mx', [`${m}${cap(l)}`, `${m}${cap(r)}`]],
  ['my', [`${m}${cap(t)}`, `${m}${cap(b)}`]],
  ['p', [p]],
  ['pt', [`${p}${cap(t)}`]],
  ['pr', [`${p}${cap(r)}`]],
  ['pb', [`${p}${cap(b)}`]],
  ['pl', [`${p}${cap(l)}`]],
  ['px', [`${p}${cap(l)}`, `${p}${cap(r)}`]],
  ['py', [`${p}${cap(t)}`, `${p}${cap(b)}`]],
])
