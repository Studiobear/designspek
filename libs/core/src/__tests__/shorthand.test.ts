import { shortHandAttr } from '../shorthand'

describe('shorthand attributes', () => {
  it('were not changed', () => {
    expect(shortHandAttr).toMatchSnapshot()
  })
})
