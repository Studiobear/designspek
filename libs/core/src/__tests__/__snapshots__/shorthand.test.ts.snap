// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shorthand attributes were not changed 1`] = `
Map {
  "m" => Array [
    "margin",
  ],
  "mt" => Array [
    "marginTop",
  ],
  "mr" => Array [
    "marginRight",
  ],
  "mb" => Array [
    "marginBottom",
  ],
  "ml" => Array [
    "marginLeft",
  ],
  "mx" => Array [
    "marginLeft",
    "marginRight",
  ],
  "my" => Array [
    "marginTop",
    "marginBottom",
  ],
  "p" => Array [
    "padding",
  ],
  "pt" => Array [
    "paddingTop",
  ],
  "pr" => Array [
    "paddingRight",
  ],
  "pb" => Array [
    "paddingBottom",
  ],
  "pl" => Array [
    "paddingLeft",
  ],
  "px" => Array [
    "paddingLeft",
    "paddingRight",
  ],
  "py" => Array [
    "paddingTop",
    "paddingBottom",
  ],
}
`;
